olive/engine/footprint.py

```diff
@@ -0,0 +1,178 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import logging
+from collections import OrderedDict
+from datetime import datetime
+from pathlib import Path
+
+from olive.common.config_utils import ConfigBase, config_json_dumps, config_json_loads
+
+logger = logging.getLogger(__name__)
+
+
+class FootprintNodeMetric(ConfigBase):
+    value: dict
```

Comment:
TODO: should consider the metrics values on different EPs/accelerators for same model.